PostgreSQL exception since the ID changed to UUID

Hi,

recently I updated reaper and noticed the schedule and repair IDs changed to a timeuuid. I'm running a build from master right now.

There is an exception when using the PostgreSQL storage:

```
ERROR  [2017-06-07 12:35:36,627] [SchedulingManagerTimer] c.s.r.s.SchedulingManager - failed managing schedule for run with id: 00000000-0000-000a-b8dc-6f0948892c1e
ERROR  [2017-06-07 12:35:36,628] [SchedulingManagerTimer] c.s.r.s.SchedulingManager - catch exception
org.skife.jdbi.v2.exceptions.UnableToExecuteStatementException: org.postgresql.util.PSQLException: ERROR: invalid input syntax for integer: "00000000-0000-000c-b8dc-6f0948892c1e" [statement:"UPDATE repair_schedule SET repair_unit_id = :re
pairUnitId, state = :state, days_between = :daysBetween, next_activation = :nextActivation, run_history = :runHistorySQL, segment_count = :segmentCount, repair_parallelism = :repairParallelism, creation_time = :creationTime, owner = :owne
r, pause_time = :pauseTime WHERE id = :id", located:"UPDATE repair_schedule SET repair_unit_id = :repairUnitId, state = :state, days_between = :daysBetween, next_activation = :nextActivation, run_history = :runHistorySQL, segment_count =
:segmentCount, repair_parallelism = :repairParallelism, creation_time = :creationTime, owner = :owner, pause_time = :pauseTime WHERE id = :id", rewritten:"/* IStoragePostgreSQL.updateRepairSchedule */ UPDATE repair_schedule SET repair_uni
t_id = ?, state = ?, days_between = ?, next_activation = ?, run_history = ?, segment_count = ?, repair_parallelism = ?, creation_time = ?, owner = ?, pause_time = ? WHERE id = ?", arguments:{ positional:{}, named:{owner:'root',followingAc
tivation:io.dropwizard.jdbi.args.JodaDateTimeArgument@4512ac38,creationTime:io.dropwizard.jdbi.args.JodaDateTimeArgument@532bfbc0,nextActivation:io.dropwizard.jdbi.args.JodaDateTimeArgument@1748b980,daysBetween:9,segmentCount:4000,runHist
orySQL:com.spotify.reaper.storage.postgresql.LongCollectionSQLTypeArgumentFactory$1@165d2d57,intensity:0.100000001,pauseTime:null,repairParallelism:com.spotify.reaper.storage.postgresql.RepairParallelismArgumentFactory$1@46fe30c8,id:com.s
potify.reaper.storage.postgresql.UuidArgumentFactory$$Lambda$60/2055717838@4a87262c,state:com.spotify.reaper.storage.postgresql.ScheduleStateArgumentFactory$1@18d11245,class:class com.spotify.reaper.core.RepairSchedule,runHistory:com.spot
ify.reaper.storage.postgresql.PostgresArrayArgumentFactory$1@6719a39d,repairUnitId:com.spotify.reaper.storage.postgresql.UuidArgumentFactory$$Lambda$60/2055717838@5056a7c6}, finder:[]}]
        at org.skife.jdbi.v2.SQLStatement.internalExecute(SQLStatement.java:1338)
        at org.skife.jdbi.v2.Update.execute(Update.java:56)
        at org.skife.jdbi.v2.sqlobject.UpdateHandler$2.value(UpdateHandler.java:68)
        at org.skife.jdbi.v2.sqlobject.UpdateHandler.invoke(UpdateHandler.java:81)
        at org.skife.jdbi.v2.sqlobject.SqlObject.invoke(SqlObject.java:212)
        at org.skife.jdbi.v2.sqlobject.SqlObject$3.intercept(SqlObject.java:127)
        at org.skife.jdbi.v2.sqlobject.CloseInternalDoNotUseThisClass$$EnhancerByCGLIB$$37045e0a.updateRepairSchedule(<generated>)
        at com.spotify.reaper.storage.PostgresStorage.updateRepairSchedule(PostgresStorage.java:456)
        at com.spotify.reaper.service.SchedulingManager.skipScheduling(SchedulingManager.java:161)
        at com.spotify.reaper.service.SchedulingManager.manageSchedule(SchedulingManager.java:147)
        at com.spotify.reaper.service.SchedulingManager.run(SchedulingManager.java:83)
        at java.util.TimerThread.mainLoop(Timer.java:555)
        at java.util.TimerThread.run(Timer.java:505)
Caused by: org.postgresql.util.PSQLException: ERROR: invalid input syntax for integer: "00000000-0000-000c-b8dc-6f0948892c1e"
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2161)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:1890)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:255)
        at org.postgresql.jdbc2.AbstractJdbc2Statement.execute(AbstractJdbc2Statement.java:560)
        at org.postgresql.jdbc2.AbstractJdbc2Statement.executeWithFlags(AbstractJdbc2Statement.java:417)
        at org.postgresql.jdbc2.AbstractJdbc2Statement.execute(AbstractJdbc2Statement.java:410)
        at org.skife.jdbi.v2.SQLStatement.internalExecute(SQLStatement.java:1327)
        ... 12 common frames omitted

```
For some reason the UuidArgumentFactory is not properly called ? I'm unsure what's happening, I'll try to dig further.

--- a/src/main/java/com/spotify/reaper/core/RepairSchedule.java
+++ b/src/main/java/com/spotify/reaper/core/RepairSchedule.java
@@ -16,7 +16,10 @@ package com.spotify.reaper.core;
 import com.google.common.collect.ImmutableList;
 import com.spotify.reaper.core.RepairSegment.State;
 import com.spotify.reaper.storage.postgresql.LongCollectionSQLType;
+
+import java.util.List;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 import org.apache.cassandra.repair.RepairParallelism;
 import org.joda.time.DateTime;
@@ -85,7 +88,11 @@ public class RepairSchedule {
    * Generic collection type would be hard to map into Postgres array types.
    */
   public LongCollectionSQLType getRunHistorySQL() {
-    return new LongCollectionSQLType(runHistory);
+    List<Long> list = runHistory
+      .stream()
+      .map(UUID::getMostSignificantBits)
+      .collect(Collectors.toList());
+    return new LongCollectionSQLType(list);
   }
 
   public int getSegmentCount() {

--- a/src/main/java/com/spotify/reaper/storage/postgresql/LongCollectionSQLType.java
+++ b/src/main/java/com/spotify/reaper/storage/postgresql/LongCollectionSQLType.java
@@ -10,13 +10,13 @@ import java.util.UUID;
  */
 public class LongCollectionSQLType {
 
-  private Collection<UUID> collection;
+  private Collection<Long> collection;
 
-  public LongCollectionSQLType(Collection<UUID> collection) {
+  public LongCollectionSQLType(Collection<Long> collection) {
     this.collection = collection;
   }
 
-  public Collection<UUID> getValue() {
+  public Collection<Long> getValue() {
     if (this.collection == null) {
       return Lists.newArrayList();
     } else {

